app/assets/javascripts/stocks.js

```diff
@@ -0,0 +1,21 @@
+var init_stock;
+
+$(document).ready(function()
+{
+  init_stock();
+})
+
+init_stock = function()
+{
+  $('#stock-search-id').on('ajax:success', function(event, data, status)
+  {
+    $('#stock-search').replaceWith(data);
+    init_stock();
+  });
+
+  $('#stock-search-id').on('ajax:error', function(event, xhr, status, error)
+  {
+    $('#stock-search-error').replaceWith('Stock not found');
+    init_stock();
+  });
+}
```

Comment:
Un évènement qui débute sur un élément du DOM va remonter jusqu'à la racine (pour autant qu'il ne soit pas annulé (`stopPropagation`))
```javascript
document.addEventListener("click", function(e) { console.log(this, e.target) })
```
Placer des évènements sur des éléments qui disparaissent est pénible et il y a de meilleures solutions: http://api.jquery.com/on/#direct-and-delegated-events

Oui, Rails c'est hip de chez hip. Donc haml, coffee, sass, etc. coffeescript.org/#try: